### PR TITLE
Missing index bug

### PR DESCRIPTION
**Pull request by fluidvanadium**
adds error msg

### CODE REVIEW

File: zingolib/src/error.rs [hunk 1]
```diff
@@ -35,6 +35,8 @@ pub enum ZingoLibError {
     /// TODO: Add Doc Comment Here!
     MissingOutputIndex(TxId),
     /// TODO: Add Doc Comment Here!
+    MissingOutputIndexInThisTx,
```
Comment (zancas) [inline zingolib/src/error.rs]: Where is the `MissingOutputIndex` variant used?

Comment (fluidvanadium) [reply zingolib/src/error.rs]: decided not to use it

File: zingolib/src/wallet/transaction_records_by_id/trait_inputsource.rs [hunk 2]
```diff
@@ -36,6 +37,11 @@ pub enum InputSourceError {
     /// Value outside the valid range of zatoshis
     #[error("Value outside valid range of zatoshis. {0:?}")]
     InvalidValue(BalanceError),
+    /// Wallet data is out of date
+    #[error(
+        "Output index data is missing! Wallet data is from an ancient version, please rescan."
```
Comment (zancas) [inline zingolib/src/wallet/transaction_records_by_id/trait_inputsource.rs]: "ancient" --> "deprecated"

**Comment by zancas** [comment]
I clicked "update branch".

**Review by zancas — APPROVED** [review]

**Pull request merged**